snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/IdRequest.java

```diff
@@ -102,21 +104,28 @@ public R execute(final C context) {
 					idGenerationTimer.start();
 
 					for (final ComponentCategory category : componentCreateRequests.keySet()) {
-						final Class<? extends SnomedComponentDocument> documentClass = getDocumentClass(category);
+						final Class<? extends SnomedDocument> documentClass = getDocumentClass(category);
 						final Collection<SnomedComponentCreateRequest> categoryRequests = componentCreateRequests.get(category);
 						
-						final Set<String> userSuppliedIds = FluentIterable.from(categoryRequests)
+						final Set<String> coreComponentIds = FluentIterable.from(categoryRequests)
 								.filter(SnomedCoreComponentCreateRequest.class)
 								.filter(request -> request.getIdGenerationStrategy() instanceof ConstantIdStrategy)
 								.transform(request -> ((ConstantIdStrategy) request.getIdGenerationStrategy()).getId())
 								.toSet();
 						
-						final Set<String> existingIds = getExistingIds(context, userSuppliedIds, documentClass);
+						final Set<String> refsetMemberIds = FluentIterable.from(categoryRequests)
+							.filter(SnomedRefSetMemberCreateRequest.class)
+							.transform(request -> request.getId())
+							.toSet();
+						
+						final Set<String> userRequestedIds = Sets.newHashSet(Iterables.concat(coreComponentIds, refsetMemberIds));
+						
+						final Set<String> existingIds = getExistingIds(context, userRequestedIds, documentClass);
 						if (!existingIds.isEmpty()) {
 							// TODO: Report all existing identifiers
 							throw new AlreadyExistsException(category.getDisplayName(), Iterables.getFirst(existingIds, null));
 						} else {
-							recorder.register(userSuppliedIds);
+							recorder.register(coreComponentIds);
```

Comment:
It turned out that we need this separation due to this line, so the original form is OK as well.